### PR TITLE
Improve LAD and min-max request handing #3

### LINKED ISSUE #3: Improve LAD and min-max request handing
Currently the [historical telemetry provider](https://github.com/akhenry/openmct-yamcs/blob/master/src/providers/historical-telemetry-provider.js#L27) issues a full request for all telemetry values within a given time range irrespective of the request strategy provided by views. It will then attempt a second "min-max" request [based on the presence of a `continuation-token` in the response](https://github.com/akhenry/openmct-yamcs/blob/master/src/providers/historical-telemetry-provider.js#L54).

Instead, for min-max requests (issued by plots) the `strategy` request option should be used to determine whether a request should be issued to the `samples` endpoint instead of the full archive.

For LAD requests (determined by the use of the `latest` strategy), the full `archive` endpoint should be used, but the `limit` parameter must be set to `1`, and the order to `desc` to ensure only the latest value within the range is returned.

### PR DESCRIPTION
**Pull request by nikhilmandlik**
### Resolves: 
closes #3 

### Author Checklist:
Check | Passed
-- | --
Changes address original issue? | Y
Unit tests included and/or updated with changes? | N/A
Command line build passes? | Y
Changes have been smoke-tested? | Y
Testing instructions included? | Y

### CODE REVIEW

File: src/providers/historical-telemetry-provider.js
Comment (jvigliotta) [inline src/providers/historical-telemetry-provider.js:47]: We're getting a lot of parameters at this point along with some defaults, would an "options" paramter work here?

File: src/providers/historical-telemetry-provider.js
Comment (jvigliotta) [inline src/providers/historical-telemetry-provider.js:56]: Nice!

**Review by jvigliotta — COMMENTED** [review]
Looking good man, looks a lot cleaner. I like that you handled both sample and regular in one. Just had one comment, may not be anything, but wanted to see what you thought. Ultimately I believe @akhenry will be the final call on this one.

File: src/providers/historical-telemetry-provider.js
Comment (jvigliotta) [reply src/providers/historical-telemetry-provider.js:56]: I like the use of convertHistory to hold either function.

**Review by akhenry — CHANGES_REQUESTED** [review]
Blocking merge so I can take a look

File: src/providers/historical-telemetry-provider.js
Comment (akhenry) [inline src/providers/historical-telemetry-provider.js:59]: 👍 Good catch

File: src/providers/historical-telemetry-provider.js
Comment (akhenry) [inline src/providers/historical-telemetry-provider.js:67]: Let's use template strings for the rest of this URL as well, just to be consistent.

**Review by akhenry — CHANGES_REQUESTED** [review]
This looks great, thanks @nikhilmandlik ! Just one *tiny* thing :) Thanks!

**Comment by akhenry** [comment]
@nikhilmandlik  Conflict, otherwise it's good to merge. Thanks!

**Pull request merged**
